draft-ietf-ppm-dap.md

```diff
@@ -1514,217 +1614,253 @@ to the next phase of the aggregation protocol.
 An `AggregationJobResp` is valid only if it satisfies the following requirement:
 
 * The Helper's `prepare_steps` MUST include exactly the same report IDs in the
-  same order as either the `report_shares` in the Leader's
-  `AggregationJobInitReq` (if this is the first round of continuation) or the
-  `prepare_steps` in the Leader's `AggregationJobContinueReq` (if this is a
-  subsequent round).
+  same order as either the `prepare_steps` in the Leader's
+  `AggregationJobInitReq` (if this is the first step of the aggregation job) or
+  `AggregationJobContinueReq` (if this is a subsequent step).
 
 [[OPEN ISSUE: consider relaxing this ordering constraint. See issue#217.]]
 
 ### Aggregate Continuation {#agg-continue-flow}
 
-In the continuation phase, the leader drives the VDAF preparation of each share
+In the continuation phase, the Leader drives the VDAF preparation of each share
 in the candidate report set until the underlying VDAF moves into a terminal
-state, yielding an output share for all Aggregators or an error. This phase may
-involve multiple rounds of interaction depending on the underlying VDAF. Each
-round trip is initiated by the leader.
+state, yielding an output share for both Aggregators or a rjection. This phase
+may involve multiple HTTP requests from Leader to Helper depending on the
+underlying VDAF.
 
 #### Leader Continuation {#aggregation-leader-continuation}
 
-The Leader begins each round of continuation for a report share based on its
-locally computed prepare message and the previous PrepareStep from the Helper.
-If PrepareStep is of type `failed`, then the leader acts based on the value of
-the `ReportShareError`:
+The Leader begins each step of aggregation continuation with a prep state
+object `state` and an outbound message `outbound` for each report in the
+candidate set.
 
-  - If the error is `ReportShareError.report_too_early`, then the leader MAY try
-    to re-send the report in a later `AggregationJobInitReq`.
-  - For any other error, the leader marks the report as failed, removes it from
-    the candidate report set and does not process it further.
+If `state == Finished(out_share)` and `outbound == None` for each report, then
```

Comment:
Right now the control flow is something like this:

```
leader init
- send an aggregate init request, handle response
- go to leader continue

leader continue
- If there is nothing to do then halt. Otherwise send aggregate continue request and handle response
- go to leader continue
```

It sounds like your suggesting something like this?

```
leader init
- send an aggregate init request, handle response
- go to leader continue

leader continue
- If there is something to do, then send aggregate continue request and handle response
- If there is nothing to do, then halt.
- go to leader continue
```

That seems fine, although maybe a bit more complicated to write in English? Up to you, please feel free to push to your own PR :D

Reply:
I wrote this up, PTAL.